service/history/shardController.go

```diff
@@ -391,13 +391,6 @@ func (i *historyShardsItem) stopEngine() {
 		i.engine = nil
 		logging.LogShardEngineStoppedEvent(i.logger, i.host.Identity(), i.shardID)
 	}
-
-	// Shutting down executionMgr will close all connections
-	// to cassandra for this engine. So, make sure to
-	// close executionMgr only after stopping the engine
-	if i.executionMgr != nil {
-		i.executionMgr.Close()
```

Comment:
Can you make the ExecutionManager not Closeable i.e. remove Close() from the interface as well ?

Reply:
I thought about that as it is little tricky right now.  Once we move all the managers be created using the persistenceClientFactory then we can remove Close on all of them.